pkg/traceql/engine_metrics.go

```diff
@@ -338,6 +338,69 @@ func (c *CountOverTimeAggregator) Sample() float64 {
 	return c.count * c.rateMult
 }
 
+// MinOverTimeAggregator it calculates the mininum value over time. It can also
+// calculate the rate when given a multiplier.
+type MinOverTimeAggregator struct {
+	firstTime       bool
+	getSpanAttValue func(s Span) float64
+	min             float64
+	rateMult        float64
+}
+
+var _ VectorAggregator = (*MinOverTimeAggregator)(nil)
+
+func NewMinOverTimeAggregator(attr Attribute) *MinOverTimeAggregator {
+	var fn func(s Span) float64
+	switch attr {
+	case IntrinsicDurationAttribute:
+		fn = func(s Span) float64 {
+			return float64(s.DurationNanos()) / float64(time.Millisecond)
```

Comment:
Should we work in nanoseconds, ms, seconds? Doing something like min_over_time(duration, s), a min-max normalization?